lib/components/textEditor/richText.js

```diff
@@ -28,181 +35,200 @@ import buttonStyles from './buttonStyle.css';
 import alignmentStyles from './aligment.css';
 import linkStyles from './linkStyles.css';
 
-class RichText extends Component {
-  linkPlugin = createLinkPlugin({
-    theme: linkStyles,
-    placeholder: 'http://…',
-  });
-
-  textAlignmentPlugin = createTextAlignmentPlugin({
-    theme: { alignmentStyles },
-  });
-
-  undoPlugin = createUndoPlugin({
-    undoContent: <UndoOutlined />,
-    redoContent: <RedoOutlined />,
-    theme: {
-      undo: buttonStyles.button,
-      redo: buttonStyles.button,
-    },
-  });
+const createEditorStateFromHTML = (html) => {
+  const blocksFromHTML = convertFromHTML(html);
 
-  toolbarPlugin = createToolbarPlugin({
-    theme: { buttonStyles, toolbarStyles },
-  });
+  return ContentState.createFromBlockArray(blocksFromHTML.contentBlocks, blocksFromHTML.entityMap);
+};
 
-  plugins = [this.toolbarPlugin, this.linkPlugin, this.textAlignmentPlugin, this.undoPlugin];
+const linkPlugin = createLinkPlugin({
+  theme: linkStyles,
+  placeholder: 'http://…',
+});
+
+const textAlignmentPlugin = createTextAlignmentPlugin({
+  theme: { alignmentStyles },
+});
+
+const undoPlugin = createUndoPlugin({
+  undoContent: <UndoOutlined />,
+  redoContent: <RedoOutlined />,
+  theme: {
+    undo: buttonStyles.button,
+    redo: buttonStyles.button,
+  },
+});
 
-  // eslint-disable-next-line require-jsdoc
-  constructor(props) {
-    super(props);
+const toolbarPlugin = createToolbarPlugin({
+  theme: { buttonStyles, toolbarStyles },
+});
 
-    this.editorWrapRef = React.createRef();
+const plugins = [toolbarPlugin, linkPlugin, textAlignmentPlugin, undoPlugin];
 
-    const editorState = createEditorStateWithText('');
+const { Toolbar } = toolbarPlugin;
 
-    this.state = {
-      editorState,
-      warningModalWasShown: false,
-    };
+const { LinkButton } = linkPlugin;
 
-    this.initialEditorContent = stateToHTML(editorState.getCurrentContent());
-  }
+const RichText = forwardRef(
+  (
+    {
+      placeholder,
+      ariaLabel,
+      spellCheck,
+      maxContentSize,
+      value,
+      defaultValue,
+      getEditorData,
+      onChange,
+    },
+    ref,
+  ) => {
+    const editorWrapRef = useRef(null);
+    const editorRef = useRef(null);
+    const modal = useRef(null);
```

Comment:
Nit: I think this could be a boolean `useState`.

Reply:
I don't see a point to use useState here because there are no renders that depend on this. We just need a variable to store a value that shows if Modal is open

Reply:
Right `useState` would be the variable for that since we do not actually need a reference to the modal. It does not have to do with renders. Something like `const [isModalOpen, setIsModalOpen] = useState(false)`. Just another hook.

Reply:
I definitely see your point though